hudi-common/src/test/java/org/apache/hudi/common/functional/TestHoodieLogFormat.java

```diff
@@ -2571,6 +2579,174 @@ public void testDataBlockFormatAppendAndReadWithProjectedSchema(
     }
   }
 
+  @Test
+  public void testCorruptBlocks() throws IOException, URISyntaxException, InterruptedException {
```

Comment:
good one. we should also add a test for w/ lot of rollback blocks. not required in this patch. but we should try and add something in a follow up when time permits